hack/cluster.sh

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env bash
+#
+# This file is part of the Kepler project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2022 The Kepler Contributors
+#
+set -eu -o pipefail
+
+# NOTE: assumes that the project root is one level up
+PROJECT_ROOT=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." >/dev/null 2>&1 && pwd)
+declare -r PROJECT_ROOT
+
+# NOTE: this allows common settings to be stored as `.env` file
+# shellcheck disable=SC1091
+[[ -f "$PROJECT_ROOT/.env" ]] && source "$PROJECT_ROOT/.env"
```

Comment:
We should define the priority for such ENV definitions, now we have at least below considerations:
1. In github action, see example [here](https://github.com/sustainable-computing-io/kepler/blob/main/.github/workflows/integration_test.yml#L101-L104)
2. add a new ".env" file like this PR mentioned.
3. In the "cluster.sh" itself, the appear sequence decides the final value.
4. The "declare" command itself, "-r" means read-only?
5. Some manual export ENVs on the test environment.

So IMO, such various of ENV definitions are not good, shall we remain the only one way, either define them in GHA, or  define them in consolidate place such as ".env"?

Reply:
This feature is intended for local development usecase only and not in CI. Hence the `.env` is ignored in git. 
Common scenario. 
`make cluster-up` bring a kind cluster but I want the cluster to be 
1. 3 nodes 
2. have grafana and prometheus installed
3. have local-dev-cluster cloned to tmp/ 

This is easily supported by this script.

Reply:
`The "declare" command itself, "-r" means read-only?` yes, it means any changes beyond this point should cause failure.